app/MMB/src/scenes/timeline/events/DepartureTimelineEvent.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import * as React from 'react';
+import idx from 'idx';
+import { graphql, createFragmentContainer } from '@kiwicom/mobile-relay';
+import { Translation } from '@kiwicom/mobile-localization';
+import { StyleSheet, Color, Text } from '@kiwicom/mobile-shared';
+
+import type { DepartureTimelineEvent as DeparturelineEventType } from './__generated__/DepartureTimelineEvent.graphql';
+import TimelineEvent from '../TimelineEvent';
+import TimelineEventIcon from '../TimelineEventIcon';
+
+type Props = {|
+  +data: DeparturelineEventType,
+|};
+
+function renderNote(duration: number): React.Element<typeof Text> {
+  const hours = Math.trunc(duration / 60);
+  const minutes = duration % 60;
+  let note;
+  if (hours === 0) {
+    note = (
+      <Translation
+        id="mmb.booking_timeline.event.departure.note_minutes"
+        values={{ minutes }}
+      />
+    );
+  } else if (hours === 1) {
+    if (minutes === 0) {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hour"
+          values={{ hours }}
+        />
+      );
+    } else if (minutes === 1) {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hour_minute"
+          values={{ hours, minutes }}
+        />
+      );
+    } else {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hour_minutes"
+          values={{ hours, minutes }}
+        />
+      );
+    }
+  } else {
+    if (minutes === 0) {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hours"
+          values={{ hours }}
+        />
+      );
+    } else if (minutes === 1) {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hours_minute"
+          values={{ hours, minutes }}
+        />
+      );
+    } else {
+      note = (
+        <Translation
+          id="mmb.booking_timeline.event.departure.note_hours_minutes"
+          values={{ hours, minutes }}
+        />
+      );
+    }
+  }
+
+  return <Text style={styles.note}>{note}</Text>;
+}
+
+const DepartureTimelineEvent = (props: Props) => {
+  const timestamp = idx(props, _ => _.data.timestamp);
+  const destination = idx(props, _ => _.data.location.airport.city.name);
+  const duration = idx(props, _ => _.data.duration);
+
+  const note = duration ? renderNote(duration) : null;
```

Comment:
this logic could live in the component, if it is not passed any duration, it could return null